compiler/src/dotty/tools/dotc/quoted/QuoteContextImpl.scala

```diff
@@ -1002,7 +1002,10 @@ class QuoteContextImpl private (ctx: Context) extends QuoteContext, scala.intern
         case _ => None
     end TypeTreeTypeTest
 
-    object TypeTree extends TypeTreeModule
+    object TypeTree extends TypeTreeModule:
+      def of[T <: AnyKind](using tp: scala.quoted.Type[T]): TypeTree =
+        tp.asInstanceOf[scala.internal.quoted.Type[TypeTree]].typeTree
```

Comment:
Why is that cast needed?

Reply:
More precisely, why is it `Type[TypeTree]` and not `Type[T]`

Reply:
Because it is the `scala.internal.quoted.Type` and not `scala.quoted.Type`. That variant is parametrised with the tree type to make it easier to cast and extract internally.